### PR TITLE
Add feature resolve conflict

### PR DESCRIPTION
**Pull request by t-cheepeng**
Task: #58

### CODE REVIEW

File: src/main/java/seedu/exercise/commons/util/AppUtil.java [hunk 1]
```diff
@@ -36,4 +37,13 @@ public static void checkArgument(Boolean condition, String errorMessage) {
             throw new IllegalArgumentException(errorMessage);
         }
     }
+
+    /**
+     * Checks that {@code state} of the program is actually {@code state}. Used for validating a particular
```
Comment (weihaw08) [inline src/main/java/seedu/exercise/commons/util/AppUtil.java]: Weird phrasing here

> Checks that {@code state} of the program is actually {@code state}

Comment (weihaw08) [reply src/main/java/seedu/exercise/commons/util/AppUtil.java]: Also why is ``assert`` used over here?

Comment (t-cheepeng) [reply src/main/java/seedu/exercise/commons/util/AppUtil.java]: will change the docs there. will change to throw exception to be more consistent with the remaining methods in there. thanks

File: src/main/java/seedu/exercise/logic/commands/ScheduleRegimeCommand.java [hunk 2]
```diff
@@ -49,16 +55,18 @@ private void checkExistenceOfRegime(Model model) throws CommandException {
 
     /**
      * Checks for scheduling conflicts and returns a valid schedule if no conflicts are found.
-     *
-     * @throws ScheduleException when there is another regime scheduled on the same date as {@code dateToSchedule}
+     * If a conflict is found, returns a null schedule
      */
-    private Schedule checkSchedulingConflict(Model model) throws ScheduleException {
+    private Schedule checkSchedulingConflict(Model model) {
```
Comment (garylyp) [inline src/main/java/seedu/exercise/logic/commands/ScheduleRegimeCommand.java]: It feels weird that after the check returns a Schedule. Perhaps you can separate out the initialization of schedule and the checking of conflicts into separate methods

File: src/main/java/seedu/exercise/commons/util/AppUtil.java (hunk 1 above)
Comment (t-cheepeng) [reply src/main/java/seedu/exercise/commons/util/AppUtil.java]: done

File: src/main/java/seedu/exercise/logic/commands/ScheduleRegimeCommand.java (hunk 2 above)
Comment (t-cheepeng) [reply src/main/java/seedu/exercise/logic/commands/ScheduleRegimeCommand.java]: changed method name to reflect what the method does better

**Pull request merged**